File: src/main/java/nl/hanze/week2/assignment45/proxy/ATMProxy.java
package nl.hanze.week2.assignment45.proxy;

import nl.hanze.week2.assignment45.ATM;
import nl.hanze.week2.assignment45.ATMState;

/**
 * Created by Laurens on 9-10-2017.
 */
public class ATMProxy implements AbstractAtm {

    private AbstractAtm realATM = new ATM();
    private User user;

    public ATMProxy(User user) {
        this.user = user;
    }

    @Override
    public ATMState getState() {
        if (user.getAge() < 10) {
            return null;
        }
        return realATM.getState();
    }

    @Override
    public int getCashInMachine() {
        if (user.getAge() < 10) {
            return -1;
        }
        return realATM.getCashInMachine();
    }
}

File: src/main/java/nl/hanze/week2/assignment45/proxy/ATMProxy.kt
package nl.hanze.week2.assignment45.proxy

import nl.hanze.week2.assignment45.ATM
import nl.hanze.week2.assignment45.ATMState

/**
 * Created by Laurens on 9-10-2017.
 */
class ATMProxy(private val user: User) : AbstractAtm {

    private val realATM = ATM()

    override val state: ATMState?
        get() = if (user.age < 10) null else realATM.state

    override val cashInMachine: Int
        get() = if (user.age < 10)  -1 else realATM.cashInMachine
}
